[MIG] crm_facebook_leads: migrate to 15.0 t#58056

luisg123v opened the pull request:
The following is performed:
- Rename field `odoo_field` on model `crm.facebook.form.field` to add
  missing `_id` suffix, as it's a many2one field
- Remove "Facebook Add" and "Facebook Campaign" from the lead model, as
  those fields always hold the same values as lead's medium and
  campaign, respectively
- Remove `.pot` file, as it's not actually being used because we handle
  translations manually

luisg123v commented:
Hi @tomeyro & @frahikLV,

Could you review, please?

@carlosrve could you review Frensh translations, please?

Regards,

CC @hugho-ad @desdelinux

frahikLV commented on crm_facebook_leads/__manifest__.py:
```diff
@@ -4,8 +4,8 @@
+    'version': '15.0.1.0.0',
     'depends': [
         'crm',
     ],
```
```suggestion
    ],
    'license': 'LGPL-3',
```

luisg123v replied:
I would like to, but I cant, because we're not the original authors of this module

frahikLV commented on crm_facebook_leads/models/crm_facebook_form.py:
```diff
@@ -48,12 +48,12 @@ class CrmFacebookFormField(models.Model):
 
     form_id = fields.Many2one('crm.facebook.form', required=True, ondelete='cascade', string='Form')
```
could you split this class in a new `.py` file?

luisg123v replied:
done

frahikLV commented on crm_facebook_leads/models/crm_facebook_form.py:
```diff
@@ -48,12 +48,12 @@ class CrmFacebookFormField(models.Model):
         'ir.model.fields',
         domain=[('model', '=', 'crm.lead'), ('store', '=', True), ('ttype', 'in', (
             'char', 'date', 'datetime', 'float', 'html', 'integer', 'monetary', 'many2one', 'selection', 'phone',
             'text'))], required=False)
```
`required=False` by default, isn't?

luisg123v replied:
Yes, done

frahikLV commented on crm_facebook_leads/i18n/es.po:
```diff
@@ -4,22 +4,34 @@
+#: model:ir.cron,cron_name:crm_facebook_leads.crm_facebook_cron
+#: model:ir.cron,name:crm_facebook_leads.crm_facebook_cron
+msgid "CRM Facebook Lead Ads: Fetch leads"
+msgstr "CRM Facebook Lead Ads: Obtener anuncios "
```
`anuncios` it's not a good translation to `leads`

```suggestion
msgstr "CRM Facebook Lead Ads: Obtener Clientes Potenciales"
```
or 
```suggestion
msgstr "CRM Facebook Lead Ads: Obtener Iniciativas"
```

luisg123v replied:
Changed by iniciativas, which is the native translation for leads

frahikLV commented on crm_facebook_leads/i18n/es.po:
```diff
@@ -51,24 +63,29 @@ msgstr "Nombre a Mostrar"
+#. module: crm_facebook_leads
+#: model:ir.model.fields,field_description:crm_facebook_leads.field_utm_medium__id_facebook_ad
+msgid "Facebook Ad ID"
+msgstr "ID de anuncio nde Facebook"
```
```suggestion
msgstr "ID de anuncio de Facebook"
```

luisg123v replied:
Done

frahikLV commented on crm_facebook_leads/i18n/es.po:
```diff
@@ -4,22 +4,34 @@
+#. module: crm_facebook_leads
+#: model:ir.model.fields,field_description:crm_facebook_leads.field_utm_adset__id_facebook_adset
+msgid "Adset ID"
+msgstr "ID de grupo de anuncios"
```
```suggestion
msgstr "ID del anuncio"
```

luisg123v replied:
Done

frahikLV commented on crm_facebook_leads/models/crm_lead.py:
```diff
@@ -11,7 +11,7 @@
+    id_facebook_lead = fields.Char(string="Facebook Lead ID", readonly=True)
     facebook_page_id = fields.Many2one(
         'crm.facebook.page', related='facebook_form_id.page_id', store=True)
     facebook_form_id = fields.Many2one('crm.facebook.form', readonly=True)
```
The fields `facebook_ad_id` and `facebook_campaign_id` are related to a field that is already in the `crm.lead` model, also, they are not being shown in the views, and I don't see any use of these fields, can you check and remove them if necessary.

luisg123v replied:
Totally correct. It doesn't make sense to have related fields to the same model, besides they're also stored.

Removed and updated commit/PR description accordingly

frahikLV reviewed (changes requested):
Hi @luisg123v can you address these changes?

luisg123v committed:
[MIG] crm_facebook_leads: migrate to 15.0

The following is performed:
- Rename field `odoo_field` on model `crm.facebook.form.field` to add
  missing `_id` suffix, as it's a many2one field
- Remove "Facebook Add" and "Facebook Campaign" from the lead model, as
  those fields always hold the same values as lead's medium and
  campaign, respectively
- Remove `.pot` file, as it's not actually being used because we handle
  translations manually

luisg123v commented:
@frahikLV could you review again, please?

frahikLV reviewed (approved):
![image](https://user-images.githubusercontent.com/47753811/172478212-3a4d634a-3275-49a0-8bff-5287c03c4fa6.png)

desdelinux commented:
LGTM
